pkg/agent/attestor/node/node_test.go

```diff
@@ -2,271 +2,418 @@ package attestor
 
 import (
 	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/tls"
 	"crypto/x509"
-	"io"
 	"io/ioutil"
+	"math/big"
 	"net/url"
 	"os"
-	"path"
+	"path/filepath"
 	"testing"
+	"time"
 
-	"github.com/golang/mock/gomock"
 	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spiffe/spire/pkg/agent/plugin/keymanager/memory"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/common/idutil"
+	"github.com/spiffe/spire/pkg/common/pemutil"
 	"github.com/spiffe/spire/pkg/common/telemetry"
 	"github.com/spiffe/spire/proto/spire/agent/keymanager"
-	"github.com/spiffe/spire/proto/spire/agent/nodeattestor"
+	agentnodeattestor "github.com/spiffe/spire/proto/spire/agent/nodeattestor"
 	"github.com/spiffe/spire/proto/spire/api/node"
 	"github.com/spiffe/spire/proto/spire/common"
+	servernodeattestor "github.com/spiffe/spire/proto/spire/server/nodeattestor"
 	"github.com/spiffe/spire/test/fakes/fakeagentcatalog"
-	mock_keymanager "github.com/spiffe/spire/test/mock/proto/agent/keymanager"
-	mock_nodeattestor "github.com/spiffe/spire/test/mock/proto/agent/nodeattestor"
-	mock_node "github.com/spiffe/spire/test/mock/proto/api/node"
-	"github.com/spiffe/spire/test/util"
-	"github.com/stretchr/testify/suite"
+	"github.com/spiffe/spire/test/fakes/fakeagentnodeattestor"
+	"github.com/spiffe/spire/test/fakes/fakeservernodeattestor"
+	"github.com/spiffe/spire/test/spiretest"
+	"github.com/stretchr/testify/require"
 )
 
 var (
-	ctx = context.Background()
+	testKey, _ = pemutil.ParseSigner([]byte(`-----BEGIN PRIVATE KEY-----
```

Comment:
consider generating this on the fly and not sharing it between the ca and agent certificates.